cajita/unit_test/tstSplineEvaluation2d.hpp

```diff
@@ -0,0 +1,448 @@
+        Kokkos::ViewAllocateWithoutInitializing( "PointSet::distance" ),
+        num_alloc );
+
+    auto local_mesh = createLocalMesh<Kokkos::HostSpace>( local_grid );
+
+    int idx_low[2] = { 0, 0 };
+    int idx_low_p1[2] = { 1, 1 };
+    double lowx[2];
+    double lowxp1[2];
+    local_mesh.coordinates( Node(), idx_low, lowx );
+    local_mesh.coordinates( Node(), idx_low_p1, lowxp1 );
+    point_set.dx = lowxp1[0] - lowx[0];
+
+    point_set.rdx = 1.0 / point_set.dx;
+
+    local_mesh.coordinates( EntityType(), idx_low,
+                            point_set.low_corner.data() );
+
+    updatePointSet( local_mesh, EntityType(), points, num_point, point_set );
+
+    return point_set;
+}
+
+//---------------------------------------------------------------------------//
+template <class DataTags>
+void splineEvaluationTest()
+{
+    // Create the global mesh.
+    std::array<double, 2> low_corner = { -1.2, 0.1 };
+    std::array<double, 2> high_corner = { -0.3, 9.5 };
+    double cell_size = 0.05;
+    auto global_mesh =
+        createUniformGlobalMesh( low_corner, high_corner, cell_size );
+
+    // Create the global grid.
+    DimBlockPartitioner<2> partitioner;
+    std::array<bool, 2> is_dim_periodic = { true, true };
+    auto global_grid = createGlobalGrid( MPI_COMM_WORLD, global_mesh,
+                                         is_dim_periodic, partitioner );
+
+    // Create a local grid.
+    int halo_width = 1;
+    auto local_grid = createLocalGrid( global_grid, halo_width );
+    auto local_mesh = createLocalMesh<TEST_DEVICE>( *local_grid );
+
+    // Create a point in the center of every cell.
+    auto cell_space = local_grid->indexSpace( Own(), Cell(), Local() );
+    int num_point = cell_space.size();
+    Kokkos::View<double* [2], TEST_DEVICE> points(
+        Kokkos::ViewAllocateWithoutInitializing( "points" ), num_point );
+    Kokkos::parallel_for(
+        "fill_points", createExecutionPolicy( cell_space, TEST_EXECSPACE() ),
+        KOKKOS_LAMBDA( const int i, const int j ) {
+            int pi = i - halo_width;
+            int pj = j - halo_width;
+            int pid = pi + cell_space.extent( Dim::I ) * pj;
+            double x[2];
+            int idx[2] = { i, j };
+            local_mesh.coordinates( Cell(), idx, x );
+            points( pid, Dim::I ) = x[Dim::I];
+            points( pid, Dim::J ) = x[Dim::J];
+        } );
+
+    // Create a point set with linear spline interpolation to the nodes.
+    auto point_set = createPointSet<DataTags>(
+        points, num_point, num_point, *local_grid, Node(), Spline<1>() );
+
+    // Check the point set data.
+    EXPECT_EQ( point_set.num_point, num_point );
+    EXPECT_FLOAT_EQ( point_set.dx, cell_size );
+    EXPECT_FLOAT_EQ( point_set.rdx, 1.0 / cell_size );
+    double xn_low[2];
+    int idx_low[2] = { 0, 0 };
+    local_mesh.coordinates( Node(), idx_low, xn_low );
+    for ( int d = 0; d < 2; ++d )
+        EXPECT_EQ( point_set.low_corner[d], xn_low[d] );
```

Comment:
Should this be `EXPECT_FLOAT_EQ`? There are a few in the 3D that are also still `EXPECT_EQ`

Reply:
yeah lets use float here - I will fix